tests/v2/validation/snapshot/snapshot_restore_k8s_upgrade_test.go

```diff
@@ -0,0 +1,102 @@
+//go:build (validation || extended) && !infra.any && !infra.aks && !infra.eks && !infra.gke && !infra.rke2k3s && !cluster.any && !cluster.custom && !cluster.nodedriver && !sanity && !stress
+
+package snapshot
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters/kubernetesversions"
+	"github.com/rancher/rancher/tests/framework/extensions/etcdsnapshot"
+
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type SnapshotRestoreK8sUpgradeTestSuite struct {
+	suite.Suite
+	session        *session.Session
+	client         *rancher.Client
+	clustersConfig *etcdsnapshot.Config
+}
+
+func (s *SnapshotRestoreK8sUpgradeTestSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *SnapshotRestoreK8sUpgradeTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	s.session = testSession
+
+	s.clustersConfig = new(etcdsnapshot.Config)
+	config.LoadConfig(etcdsnapshot.ConfigurationFileKey, s.clustersConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+}
+
+func (s *SnapshotRestoreK8sUpgradeTestSuite) TestSnapshotRestoreK8sUpgrade() {
+	snapshotRestoreNone := &etcdsnapshot.ETCD{
+		SnapshotRestore: "none",
+	}
+
+	snapshotRestoreK8sVersion := &etcdsnapshot.ETCD{
+		SnapshotRestore: "KubernetesVersion",
+	}
+
+	snapshotRestoreAll := &etcdsnapshot.ETCD{
+		SnapshotRestore: "all",
+	}
+
+	tests := []struct {
+		name            string
+		snapshotRestore *etcdsnapshot.ETCD
+		client          *rancher.Client
+	}{
+		{"Restore etcd only", snapshotRestoreNone, s.client},
+		{"Restore Kubernetes version and etcd", snapshotRestoreK8sVersion, s.client},
+		{"Restore cluster config, Kubernetes version and etcd", snapshotRestoreAll, s.client},
+	}
+
+	for _, tt := range tests {
+		availableVersions, err := kubernetesversions.Default(s.client, clusters.RKE2ClusterType.String(), nil)
```

Comment:
You need to get k3s versions as well not just rke2. And also what is the point of K3SKubernetesVersions and RKE2KubernetesVersions if you're only getting one item?